adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakSpringBootConfigResolver.java

```diff
@@ -21,13 +21,15 @@
 import org.keycloak.adapters.KeycloakDeploymentBuilder;
 import org.keycloak.adapters.OIDCHttpFacade;
 import org.keycloak.representations.adapters.config.AdapterConfig;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
 public class KeycloakSpringBootConfigResolver implements org.keycloak.adapters.KeycloakConfigResolver {
 
     private KeycloakDeployment keycloakDeployment;
 
+    @Autowired
```

Comment:
Can we set `required=false` so that we still keep the current behavior where the adapter config is set directly?

Reply:
Another - maybe better solution would be to replace the Autowired annotation with a constructor and an argument AdapterConfig, like this:

    public KeycloakSpringBootConfigResolver(AdapterConfig config) {
      this.adapterConfig = config;
    }

Because then, Spring will also identify, that AdapterConfig must be initialized before initializing KeycloakSpringBootConfigResolver. Because I'm not sure, if required=false really helps. But constructor injection would be fine too.

If you think, that would be better, let me know and I change the PR.

Reply:
You probably know Spring better than me :) The only reason I did that comment is that we have a previous fix that was about  making the auto-configuration work across both SB and Spring Security.

By using `required=false`, I would assume that we would still support cases where the adapter config is injected directly as it stands today. Thus, without impact (again and again) people using Spring with Keycloak.

Reply:
Ah, sorry for my late response. Good point. I'd like to try that. Because of my vacation for two weeks, I need to postpone that til mid of July. Is that okay? Also I say, that field injection with AutoWired is not best practice but constructor injection is better. But in that case, the class signature changes, what I would like to avoid. 

Also - another point - we could create a public setAdapterConfig() or replace the private field with a protected field. In both cases the Spring application which is WAR-deployment-aware can set the AdapterConfig explicitly.

But first, I'll try `required=false`

Reply:
I've tested required=false in my workspace and there it works for JAR an WAR deployment. In both cases the PR works for me.